website/static/js/paginator.es6.js

```diff
@@ -0,0 +1,122 @@
+var ko = require('knockout');
+var oop = require('js/oop');
+
+var MAX_PAGES_ON_PAGINATOR = 7;
+var MAX_PAGES_ON_PAGINATOR_SIDE = 5;    
+
+var Paginator = oop.defclass({
+    constructor() {
+        this.numberOfPages = ko.observable(0);
+        this.currentPage = ko.observable(0);
+        this.paginators = ko.observableArray([]);
+        if (!this.search){
```

Comment:
@samchrisinger @chennan47 This doesn't quite work. The constructor may be executed before the class's `search()` method is defined. How about this instead:

```js
var Paginator = oop.defclass({
    // ...
    search() {
        throw new Error('Paginator subclass must define a "search" method.');
    }
});